### PR TITLE
6803 Update CF content on Search Results page

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As a Veteran searching for institutions using the GI Bill Comparison Tool, I need to know what types of caution flags the institutions I'm considering have so that I can make the best decision about my education choices.

[Originating Issue](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/6803)

## Testing done
Testing Passes Locally

## Acceptance criteria
- [x] 1. When search results are displayed for schools that contain caution flags that are NOT for school closings:
  - [x] a. The caution flag is displayed with a header that says either "This school has a cautionary warning" or "This school has cautionary warnings".
  - [x] b. The caution flag includes brief subtext below the header that describes the caution flag in more detail.
  - [x] c. If a school has multiple caution flags then subtext bullets are included below that describe each of the caution flags.
  - [x] d. If there are multiple caution flags, they are displayed in alphabetical order based on first letter of the flag subtext that is displayed.

- [x] 2. When search results are displayed institutions with a school closing:
  - [x] a. The caution flag is displayed with a header that states: "A campus is closing soon"
  - [x] b. The caution flag includes a brief subtext below the header that describes the caution flag in more detail: "A campus at this school will be closing soon"

- [x] 3. When an institution has both a school closing and caution flags both are displayed in separate alerts with the school closing above the caution flag:

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/utils/render.jsx
Comment (zurbergram) [inline src/applications/gi/utils/render.jsx]: why not check if `cautionFlags` is empty and remove use of this attribute?
Since in https://github.com/department-of-veterans-affairs/gibct-data-service/pull/612 this attribute just becomes an empty check against the array and eventually we might want to remove `caution_flag` and `caution_flag_reason` as attributes being returned from GIDs

Comment (elder-joseph) [reply src/applications/gi/utils/render.jsx]: I had thought about doing that actually, that way we dont have to pass it over either. I'll go ahead and make that switch.

File: src/applications/gi/components/search/SearchResult.jsx
Comment (zurbergram) [inline src/applications/gi/components/search/SearchResult.jsx]: unused variable

File: src/applications/gi/utils/render.jsx
Comment (zurbergram) [inline src/applications/gi/utils/render.jsx]: Is switching this to an `li` tag going to cause axe/508 issues?

File: src/applications/gi/utils/render.jsx
Comment (zurbergram) [inline src/applications/gi/utils/render.jsx]: Should this be wrapped in `ul` tag?

File: src/applications/gi/utils/render.jsx
Comment (zurbergram) [inline src/applications/gi/utils/render.jsx]:
```suggestion
  if (cautionFlags.length > 0) return null;
```

File: src/applications/gi/utils/render.jsx
Comment (zurbergram) [inline src/applications/gi/utils/render.jsx]:
```suggestion
      isVisible={cautionFlags.length > 0}
```

File: src/applications/gi/utils/render.jsx
Comment (elder-joseph) [reply src/applications/gi/utils/render.jsx]: Just ran an Axe analyzation and had one finding, which is related to your comment below about wrapping in a `<ul>`

Comment (zurbergram) [reply src/applications/gi/utils/render.jsx]: Based on images in story I think you can leave as a `<p>`

File: src/applications/gi/utils/render.jsx
Comment (shawkey-daniel) [inline src/applications/gi/utils/render.jsx]: Won't this return null, if there are caution flags?

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

File: src/applications/gi/utils/render.jsx
Comment (elder-joseph) [reply src/applications/gi/utils/render.jsx]: After talking with Matt, pushed a fix for this here.

**Review by zurbergram — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

**Review by rianfowler — APPROVED** [review]

**Comment by elder-joseph** [comment]
Hey @rianfowler,

I saw your approval but added in an additional check in case the `cautionFlags` array isn't present. Didnt want to sneak that in without letting you know.

**Comment by rianfowler** [comment]
thanks for the heads up @elder-joseph . Since this only changes app specific code, you can configure github to skip the required review by the @department-of-veterans-affairs/frontend-review-group by adding your team as a code owner over this app directory: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/codeowners.md

File: src/applications/gi/components/vet-tec/VetTecProgramSearchResult.jsx
Comment (bkjohnson) [inline src/applications/gi/components/vet-tec/VetTecProgramSearchResult.jsx]: For this additional check you could have used [optional chaining](https://dsva.slack.com/archives/CBU0KDSB1/p1567182149038900)

**Pull request merged**
